packages/devtools_app/lib/src/config_specific/import_export/import_export.dart

```diff
@@ -71,23 +71,20 @@ class ImportController {
     }
     previousImportTime = now;
 
-    final isDevToolsSnapshot =
-        json is Map<String, dynamic> && json[devToolsSnapshotKey] == true;
-    if (!isDevToolsSnapshot) {
+    final devToolsSnapshot = json;
+    if (devToolsSnapshot is Map<String, dynamic> &&
+        devToolsSnapshot[devToolsSnapshotKey] == true) {
+      final activeScreenId = devToolsSnapshot[activeScreenIdKey];
+      offlineController
+        ..enterOfflineMode()
+        ..offlineDataJson = devToolsSnapshot;
+      serviceManager.connectedApp =
+          OfflineConnectedApp.parse(devToolsSnapshot[connectedAppKey]);
+      _notifications.push(attemptingToImportMessage(activeScreenId));
+      _pushSnapshotScreenForImport(activeScreenId);
+    } else {
       _notifications.push(nonDevToolsFileMessage);
```

Comment:
I like two things in original code, that are lost in new code:

1. It follows 'early out' (or 'return early') pattern by returning immediately when it is clear the path is non-happy.
More materials on this:
https://szymonkrajewski.pl/why-should-you-return-early/
https://g3doc.corp.google.com/company/teams/gpay/flutter/development/code_style.md#prefer-early-out-style

2. It defines the variable isDevToolsSnapshot, that described the meaning of the logical expression with its name.

From my perspective the original code is more readable, than the new one.
What do you think?

Reply:
Oh yes, I agree. This change wasn't about readability, but about getting rid of the "unnecessary cast" warning in g3. This way both the internal and external linters are happy without suppressing any warnings. The analyzer doesn't seem smart enough to know the type of `devToolsSnapshot` with the early out pattern.

Reply:
Just to clarify, explicitly setting the cast is what was there before, right?

Reply:
Ok suppressing works. I'll just add that change in g3!